docs/commands.md

```diff
@@ -0,0 +1,87 @@
+---
+id: commands
+title: "Commands"
+---
+
+`Command[Model]` is a description of a CLI command. This is parsed by `CliApp` to produce a collection of valid commands and transform a valid input from a user into an instance of type `Model`. Then you can implement the functionality of the CLI App using pattern matching on instances of `Model`.
+
+
+
+It's possible to create a `Command` using the `apply` method. You must always specify the name of the command that will be used
+in the CLI, but you can also specify options and/or arguments:
+
+```scala mdoc:silent
+import zio.cli._
+
+object Command {
+  def apply(name: String) = ???
+  def apply[ArgsType](name: String, args: Args[ArgsType]) = ???
+  def apply[OptionsType](name: String, options: Options[OptionsType]) = ???
+  def apply[OptionsType, ArgsType](name: String, options: Options[OptionsType], args: Args[ArgsType]) = ???
+}
+```
+## Combining commands and transformations
+**ZIO CLI** also allows to combine different commands in a functional manner.
+
+```scala mdoc:silent
+trait Command[+A] {
+  def map[B](f: A => B): Command[B]
+  def |[A1 >: A](that: Command[A1]): Command[A1]
+  def orElse[A1 >: A](that: Command[A1]): Command[A1]
+  def orElseEither[B](that: Command[B]): Command[Either[A, B]]
+  def withHelp(help: HelpDoc): Command[A]
+  def withHelp(help: String): Command[A]
+  def subcommands[B](that: Command[B])
+  def subcommands[B](c1: Command[B], c2: Command[B], cs: Command[B]*)
+}
+```
+### Example
+In the following example, we are going to construct a command using some of the methods above
+```scala mdoc:silent:reset
+import zio.cli._
+
+// Construction of basic commands and HelpDoc
+val parentCommand: Command[BigInt] = Command("pC", Options.integer("int"))
```

Comment:
As before, instead of generic examples like BigInt, String, etc., try to pick realistic commands based on existing command-line applications.